src/dot.rs

```diff
@@ -171,10 +171,13 @@ impl<W> fmt::Write for Escaper<W>
 
     fn write_char(&mut self, c: char) -> fmt::Result {
         match c {
-            '"' => try!(self.0.write_char('\\')),
+            '"' | '\\' | '|' | '<' | '>' | '&' => try!(self.0.write_char('\\')),
             // \l is for left justified linebreak
-            '\n' => return self.0.write_str(r#"\l"#),
-            _   => { }
+            '\n' => {
+                try!(self.0.write_char('\\'));
+                return self.0.write_char('l');
```

Comment:
Why did you change from write_str to 2 write_chars here? I guess the difference is trivial, but normally write_str should be better w.r.t code quality.

Reply:
I had double escaping issues at some point because I was using `self.write_str`, but I think it might be fine if I use `self.0.write_str`.